crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -1076,6 +1088,50 @@ macro_rules! timing_trace_out {
     };
 }
 
+/// Select the first successful response that has non-empty data.
+///
+/// Await all of the futures at once and return one of (in order of priority):
+/// 1. The first valid, non-empty response, as deemed by the `is_empty` function.
+/// 2. The last valid but empty response, once all futures have completed.
+/// 3. The last error if none of the futures produce a valid response.
+///
+/// Note: If one or more future does not produce a response, thus timeout, and one or more produces
```

Comment:
It may be only me that stumbles over this, so take or leave as preferred.
```suggestion
/// Note: If one or more futures do not produce a response, thus time out, and one or more produce
```